main.cc

```diff
@@ -1146,6 +1153,7 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
             if (udf_enabled) {
                 qp.invoke_on_all([&] (cql3::query_processor& qp) {
                     qp.set_wasm_instance_cache(&wasm_instance_cache.local());
+                    qp.set_alien_runner(alien_runner);
```

Comment:
What prevents use of the alien runner before it is set?

Reply:
The code structure - similarly to the instance cache - the runner's only used for initialization when parsing system tables (which happens later in the scylla_main()) and when executing queries (which is started even later in scylla_main()).
We could add the runner(and cache) to the query_processor constructor, but the wasm parameters are sill optional (UDFs are experimental) and it's already taking a lot of parameters so I don't know if we want it. We could also add some checks when trying to use them (alien runner and cache), but it seems like it's simpler to just keep the order of operations correct.

Reply:
I think it's nicer to pass it as a constructor parameter, it eliminates the question about use-before-set. It's an idiom in C++, pass your dependencies up-front.

UDF will not be experimental forever (in fact, I hope we can move them out of experimental soon).

Reply:
I did not add this in the last rebase, but I assume we would like to do this also for the instance cache and also wasmtime::Engine (which is currently in the `database`)? If so, maybe I'll just make a follow-up, which moves them all to the constructor

Reply:
ok for a follow-up.